src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -862,13 +865,13 @@ public static TypeConverter GetConverter([DynamicallyAccessedMembers(Dynamically
         /// Returns a custom type descriptor for the given type.
         /// Performs arg checking so callers don't have to.
         /// </summary>
-        internal static ICustomTypeDescriptor? GetDescriptor(
+        private static DefaultTypeDescriptor GetDescriptor(
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type type,
             string typeName)
         {
-            ArgumentNullException.ThrowIfNull(type);
+            ArgumentNullException.ThrowIfNull(type, typeName);
 
-            return NodeFor(type).GetTypeDescriptor(type);
+            return NodeFor(type).GetDefaultTypeDescriptor(type);
```

Comment:
The code in this new GetDefaultTypeDescriptor is what you would have gotten from the GetTypeDescriptor override that was there before?

Reply:
Correct. 

`NodeFor(type)` always returns a private class `TypeDescriptionNode`.

https://github.com/dotnet/runtime/blob/62eb291312749c52507309d50051dd61538cc62d/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs#L1425

`GetTypeDescriptor(type)` is a non-virtual method on the base `TypeDescriptionProvider` class that forwards to the `GetTypeDescriptor(type, instance)` virtual method:

https://github.com/dotnet/runtime/blob/62eb291312749c52507309d50051dd61538cc62d/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptionProvider.cs#L217-L220

Which is overriden by `TypeDescriptionNode` to always return a `DefaultTypeDescriptor`:

https://github.com/dotnet/runtime/blob/62eb291312749c52507309d50051dd61538cc62d/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs#L3107-L3117

I wanted to do this for the `GetConverter` method that takes an object instance as well, but couldn't because there it can return either a `DefaultTypeDescriptor` or a `MergedTypeDescriptor`, so I had to continue letting the helper method return `ICustomTypeDescriptor` and added the assert that whomever we called returned non-null from `GetConverter()`.